### PR TITLE
Print port number when adding new interface to switch

### PR DESCRIPTION
**Pull request by kmateuszssak**

### CODE REVIEW

File: CLI/pipeline.c [hunk 1]
```diff
@@ -20,11 +20,19 @@
...
 #include <string.h>
+#include <jansson.h>
 
 #include "../include/psabpf.h"
 #include "../include/psabpf_pipeline.h"
 #include "common.h"
 
+static void print_port(const char *intf, int ifindex) {
+    json_t *root = json_object();
+    json_object_set(root, "name", json_string(intf));
```
Comment (tatry) [inline CLI/pipeline.c:31]: Use `json_object_set_new` here and below to avoid memory leak

File: include/psabpf_pipeline.h [hunk 2]
```diff
@@ -21,5 +21,5 @@ bool psabpf_pipeline_exists(psabpf_context_t *ctx);
 /* This function should load BPF program and initialize default maps (call map initializer program) */
 int psabpf_pipeline_load(psabpf_context_t *ctx, const char *file);
 int psabpf_pipeline_unload(psabpf_context_t *ctx);
-int psabpf_pipeline_add_port(psabpf_context_t *ctx, const char *interface);
+int psabpf_pipeline_add_port(psabpf_context_t *ctx, const char *interface, int *ifindex);
```
Comment (tatry) [inline include/psabpf_pipeline.h:24]: I suggest `port_id` instead of `ifindex`

File: lib/psabpf_pipeline.c [hunk 3]
```diff
@@ -389,13 +389,13 @@ int psabpf_pipeline_add_port(psabpf_context_t *ctx, const char *interface)
     build_ebpf_prog_filename(pinned_file, sizeof(pinned_file), ctx, XDP_HELPER_PROG);
     isXDP = access(pinned_file, F_OK) != 0;
 
-    int ifindex = (int) if_nametoindex(interface);
-    if (!ifindex) {
+    *ifindex = (int) if_nametoindex(interface);
```
Comment (tatry) [inline lib/psabpf_pipeline.c:392]: I think this argument should be optional, I mean `if (ifindex != NULL) *ifindex = ...;`

**Review by tatry — CHANGES_REQUESTED** [review]
Three small questions

File: lib/psabpf_pipeline.c (hunk 3 above)
Comment (kmateuszssak) [reply lib/psabpf_pipeline.c]: `if_nametoindex` returns 0 if tere is no such interface. 
https://man7.org/linux/man-pages/man3/if_nametoindex.3.html

Comment (tatry) [reply lib/psabpf_pipeline.c]: I mean that pointer to result should be checked for NULL-ness. `if_nametoindex ` have nothing to the address of that variable.

**Review by tatry — APPROVED** [review]

**Pull request merged**